src/metpy/plots/declarative.py

```diff
@@ -1232,10 +1297,16 @@ def _set_need_rebuild(self, _):
 
     def _build(self):
         """Build the plot by calling any plotting methods as necessary."""
-        x, y, imdata = self.plotdata
-        self.handle = self.parent.ax.contourf(x, y, imdata, self.contours,
+        x_like, y_like, imdata = self.plotdata
+
+        if isinstance(self.parent, MapPanel):
```

Comment:
I think the right way to tackle this in an OO framework is to create some kind of property on `Panel` and `MapPanel` that has this, like `plot_kwargs`.